pkg/icons/icons.go

```diff
@@ -0,0 +1,80 @@
+package icons
+
+import (
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/rancher/charts-build-scripts/pkg/filesystem"
+	"github.com/rancher/charts-build-scripts/pkg/path"
+	"github.com/sirupsen/logrus"
+	helmRepo "helm.sh/helm/v3/pkg/repo"
+)
+
+// DownloadIcons download icons to use locally
+func DownloadIcons(rootFs billy.Filesystem) error {
+	absRepositoryHelmIndexFile := filesystem.GetAbsPath(rootFs, path.RepositoryHelmIndexFile)
+	helmIndexFile, err := helmRepo.LoadIndexFile(absRepositoryHelmIndexFile)
+	if err != nil {
+		return fmt.Errorf("encountered error while trying to load existing index file: %s", err)
+	}
+	helmIndexFile.SortEntries()
+	for name := range helmIndexFile.Entries {
+		latestEntryVersion := helmIndexFile.Entries[name][0]
+		latestIconName := name
+		if strings.Contains(name, "-crd") {
```

Comment:
// We don't add any icons for "CRD" because they are not displayed as installable charts